### PR TITLE
add dish-description-api and unit test

### PR DESCRIPTION
**Pull request by SoniaChoo**

### CODE REVIEW

File: db/dish.go [hunk 1]
```diff
@@ -107,3 +107,32 @@ func SearchByDishNameDish(d *model.Dish) ([]*model.Dish, error) {
...
+	if err != nil {
+		log.Printf("error connect database, %v\n", err)
+		return nil, err
+	}
+
+	//start to excute SQL query
+	dishs := []*model.Dish{}
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	row, err := db.QueryContext(ctx, "select * from dish where  description like ?", "%"+d.Description+"%")
```
Comment (ziyitony) [inline db/dish.go]: please refrain from making too much space.

Comment (SoniaChoo) [reply db/dish.go]: updated.

File: http/dishHandler.go [hunk 2]
```diff
@@ -178,5 +181,44 @@ func SearchByDishNameHandlerDish(w http.ResponseWriter, r *http.Request) {
...
+	var dish model.Dish
+	if err = json.Unmarshal(reqBytes, &dish); err != nil {
+		log.Printf("Read dish info error! Error is %s\n", err.Error())
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, BadJsonDishSearch)
+		return
+	}
+
+	//check if the request is no sense
+	if strings.Trim(dish.Description, " ") == "" {
```
Comment (ziyitony) [inline http/dishHandler.go]: please use `strings.TrimSpace()`, and update `description` itself.

**Review by ziyitony — APPROVED** [review]

File: http/dishHandler.go (hunk 2 above)
Comment (SoniaChoo) [reply http/dishHandler.go]: updated

**Pull request merged**